### PR TITLE
codelens support

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
totally async and doesn't depend on the pipeline thread 😃

### CODE REVIEW

File: src/PowerShellEditorServices.Engine/Services/CodeLens/CodeLensData.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+namespace PowerShellEditorServices.Engine.Services.CodeLenses
```
Comment (rjmholt) [inline src/PowerShellEditorServices.Engine/Services/CodeLens/CodeLensData.cs]: Namespace

File: src/PowerShellEditorServices.Engine/Services/CodeLens/CodeLensData.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
```
Comment (rjmholt) [inline src/PowerShellEditorServices.Engine/Services/CodeLens/CodeLensData.cs]: Copyright

File: src/PowerShellEditorServices.Engine/PowerShellEditorServices.Engine.csproj [hunk 3]
```diff
@@ -21,4 +21,7 @@
     <PackageReference Include="System.Security.Principal.Windows" Version="4.5.1" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Folder Include="Services\CodeLens\" />
```
Comment (rjmholt) [inline src/PowerShellEditorServices.Engine/PowerShellEditorServices.Engine.csproj]: What does this entry do? Is there a reason CodeLens requires it when other features/services don't?

File: src/PowerShellEditorServices.Engine/Services/CodeLens/IScriptExtentExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System.Management.Automation.Language;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Microsoft.PowerShell.EditorServices
+{
+    public static class IScriptExtentExtensions
```
Comment (rjmholt) [inline src/PowerShellEditorServices.Engine/Services/CodeLens/IScriptExtentExtensions.cs]: Worth making internal?

File: src/PowerShellEditorServices.Engine/PowerShellEditorServices.Engine.csproj (hunk 3 above)
Comment (TylerLeonhardt) [reply src/PowerShellEditorServices.Engine/PowerShellEditorServices.Engine.csproj]: I think it was VS that added it. I'll delete it

File: src/PowerShellEditorServices.Engine/Services/CodeLens/IScriptExtentExtensions.cs (hunk 4 above)
Comment (TylerLeonhardt) [reply src/PowerShellEditorServices.Engine/Services/CodeLens/IScriptExtentExtensions.cs]: done

File: src/PowerShellEditorServices.Engine/Services/CodeLens/CodeLensData.cs (hunk 2 above)
Comment (TylerLeonhardt) [reply src/PowerShellEditorServices.Engine/Services/CodeLens/CodeLensData.cs]: added

File: src/PowerShellEditorServices.Engine/Services/CodeLens/CodeLensData.cs (hunk 1 above)
Comment (TylerLeonhardt) [reply src/PowerShellEditorServices.Engine/Services/CodeLens/CodeLensData.cs]: changed

**Pull request merged**